src/include/migraphx/target_assignments.hpp

```diff
@@ -35,6 +35,10 @@ struct target_assignments
 {
     void add_assignment(instruction_ref ins, const std::string& target);
 
+    auto size() const { return assignments.size(); }
+    auto& at(instruction_ref ins) const { return assignments.at(ins); }
+    bool has(instruction_ref ins) const { return assignments.find(ins) != assignments.end(); }
```

Comment:
You should just expose `find` so we can just use `contains` function. Also this will make it easier to swap out with a map in the future if we want to.